store/sqlstore/channel_store.go

```diff
@@ -1617,6 +1617,8 @@ func (s SqlChannelStore) buildFulltextClause(term string) (fulltextClause, fullt
 
 	// Prepare the FULLTEXT portion of the query.
 	if s.DriverName() == model.DATABASE_DRIVER_POSTGRES {
+		fulltextTerm = strings.Replace(fulltextTerm, "|", "", -1)
```

Comment:
I think there are more characters than just `|` that have a special meaning in a postgres tsquery - and you could actually escape them with a backslash instead of removing them altogether.

Looking at some code of mine that uses tsquery, I'm prefixing all these characters with a backslash: `@<>!()&|:'` - but probably best to check the postgres docs what really needs to be escaped.

Reply:
This comment has reminded me that I think somewhere in the posts search code we have a list of characters that get escaped for each SQL database server. Maybe we can re-use that here (and add any missing chars to it at the same time)?

Reply:
I think it depends a lot on the context - dynamic SQL (I hope MM isn't doing that) requires different escaping than a `LIKE` statement or a tsquery.

Reply:
Yes, we already have that but @ThiefMaster is correct that it depends on the context. I can see if any of that is reusable in this case and add some more tests to make sure those characters are covered as well